crates/snp_validation/src/lib.rs

```diff
@@ -288,94 +310,129 @@ impl Snp {
 
         // Should only be version 2
         if report.body.version != 2 {
-            return Err(anyhow!("snp report is an unknown version"));
+            bail!("snp report is an unknown version");
         }
```

Comment:
If we're refactoring this already, then use `ensure!` here and all the other trivial conditionals in this file

Reply:
Fixed